debezium-core/src/main/java/io/debezium/converters/CloudEventsConverter.java

```diff
@@ -0,0 +1,549 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.converters;
+
+import static io.debezium.converters.SerializerType.withName;
+import static org.apache.kafka.connect.data.Schema.Type.STRUCT;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Schema.Type;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.json.JsonConverterConfig;
+import org.apache.kafka.connect.json.JsonDeserializer;
+import org.apache.kafka.connect.storage.Converter;
+import org.apache.kafka.connect.storage.ConverterConfig;
+import org.apache.kafka.connect.storage.ConverterType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import io.debezium.annotation.VisibleForTesting;
+import io.debezium.config.Configuration;
+import io.debezium.config.Instantiator;
+import io.debezium.data.Envelope;
+import io.debezium.util.SchemaNameAdjuster;
+
+/**
+ * Implementation of Converter that express schemas and objects with CloudEvents specification. The serialization
+ * format can be Json or Avro.
+ * <p>
+ * The serialization format of CloudEvents is configured with
+ * {@link CloudEventsConverterConfig#CLOUDEVENTS_SERIALIZER_TYPE_CONFIG cloudevents.serializer.type} option.
+ * <p>
+ * The serialization format of the data attribute in CloudEvents is configured with
+ * {@link CloudEventsConverterConfig#CLOUDEVENTS_DATA_SERIALIZER_TYPE_CONFIG cloudevents.data.serializer.type} option.
+ * <p>
+ * If the serialization format of the data attribute in CloudEvents is JSON, it can be configured whether to enable
+ * schemas with {@link CloudEventsConverterConfig#CLOUDEVENTS_JSON_SCHEMAS_ENABLE_CONFIG json.schemas.enable} option.
```

Comment:
```suggestion
 * schemas with {@code json.schemas.enable} option.
```

Reply:
Actually this is obsolete, as there's no dedicate option any longer, but it's replaced by allowing for generic pass-through of underlying converter options. I'll update.